webapp/channels/src/components/admin_console/ldap_wizard/ldap_wizard.tsx

```diff
@@ -242,6 +244,22 @@ const LDAPWizard = (props: Props) => {
         );
     };
 
+    const buildExpandableSetting = (setting: AdminDefinitionSetting) => {
+        return (
+            <LDAPExpandableSetting
+                key={schema.id + '_expandable_' + setting.key}
+                schema={schema}
+                setting={setting}
+                buildSettingFunction={(subSetting: LDAPDefinitionSetting) => {
```

Comment:
Why pass this in instead of having this be part of the component itself?

Reply:
Oh, I understand now that it's because of the build functions here. This makes me think more that we should try some more to separate these `buildXSetting` functions from the `LdapWizard` so that the logic that maps a setting schema to the component can be pulled out into a reusable function for cases like this

Reply:
Yep, I'm open to it -- that's the point of waiting to see what naturally deserves to be abstracted or compressed. :) So far I'd still argue that the builder functions access so much internal state that it would cost more to separate than it costs (in complexity) to keep them within the ldap wizard.  But that can always change.

Reply:
I've got some ideas for how we might redo how we store this state using a series of hooks which would this to be composed and typed a lot nicer, but I need some time to play with that before I'd be comfortable inflicting that on anyone. I think these massive state objects with arbitrary fields are something I'm responsible for adding, so it's one of the many things I want to revisit to make our lives easier 😅